### PR TITLE
Add composed device model storage and everest device model storage.

### PR DESCRIPTION
**Pull request by maaikez**
## Describe your changes

Created composed device model storage and everest device model storage like stated in https://github.com/EVerest/everest-core/tree/doc/config-service/doc/config_service#next-steps-to-adjust-ocpp201-and-libocpp

## Issue ticket number and link

#844 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Review by Pietfried — CHANGES_REQUESTED** [review]
This looks good!

I would like to see at least some parts of [this](https://github.com/EVerest/everest-core/tree/doc/config-service/doc/config_service#next-steps-to-adjust-ocpp201-and-libocpp) documented here as part of the OCPP201 module documentation:
* Class diagram + explaination
* Variable attributes and variable characteristics for externally managed variables
* Sequence of variable access for internally and externally managed variables

**Comment by maaikez** [comment]
> This looks good!
> 
> I would like to see at least some parts of [this](https://github.com/EVerest/everest-core/tree/doc/config-service/doc/config_service#next-steps-to-adjust-ocpp201-and-libocpp) documented here as part of the OCPP201 module documentation:
> 
>     * Class diagram + explaination
> 
>     * Variable attributes and variable characteristics for externally managed variables
> 
>     * Sequence of variable access for internally and externally managed variables

ok, in doc.rst? Or where else?

**Comment by Pietfried** [comment]
> > This looks good!
> > I would like to see at least some parts of [this](https://github.com/EVerest/everest-core/tree/doc/config-service/doc/config_service#next-steps-to-adjust-ocpp201-and-libocpp) documented here as part of the OCPP201 module documentation:
> > ```
> > * Class diagram + explaination
> > 
> > * Variable attributes and variable characteristics for externally managed variables
> > 
> > * Sequence of variable access for internally and externally managed variables
> > ```
> 
> ok, in doc.rst? Or where else?

Yep :+1:

**Comment by maaikez** [comment]
> > > This looks good!
> > > I would like to see at least some parts of [this](https://github.com/EVerest/everest-core/tree/doc/config-service/doc/config_service#next-steps-to-adjust-ocpp201-and-libocpp) documented here as part of the OCPP201 module documentation:
> > > ```
> > > * Class diagram + explaination
> > > 
> > > * Variable attributes and variable characteristics for externally managed variables
> > > 
> > > * Sequence of variable access for internally and externally managed variables
> > > ```
> > 
> > 
> > ok, in doc.rst? Or where else?
> 
> Yep 👍

Ok, added!

**Comment by maaikez** [comment]
Is there still a change requested?

File: modules/OCPP201/doc.rst [hunk 1]
```diff
@@ -334,3 +334,94 @@ the `external_energy_limits` interface. In addition, the config parameter `Compo
...
+* Internally Managed: Owned, stored and accessed in libocpp in device model storage
+  Examples: HeartbeatInterval, AuthorizeRemoteStart, SampledDataTxEndedMeasurands, AuthCacheStorage
+* Externally Managed: Owned, stored and accessed via EVerest config service (not yet supported)
+  Examples: ConnectionTimeout, MasterPassGroupId
+* For externally managed variables a mapping to the EVerest configuration parameter is defined (not yet supported)
+
+Note that the EVerest config service is not yet implemented. Currently all components and variables are controlled
+by the libocpp device model storage implementation.
+
+Device Model Implementation this module
```
Comment (Pietfried) [inline modules/OCPP201/doc.rst:373]:
```suggestion
Device Model Implementation of this module
```

File: modules/OCPP201/doc.rst [hunk 2]
```diff
@@ -334,3 +334,94 @@ the `external_energy_limits` interface. In addition, the config parameter `Compo
...
+
+Clarification of the device model classes of this diagram:
+
+* DeviceModel:
+
+  * Part of libocpp
+  * Contains device model representation and business logic to prevalidate requests to the device model variables
+  * Contains reference to device model interface implementation
+
+* DeviceModelInterface:
```
Comment (Pietfried) [inline modules/OCPP201/doc.rst:408]: DeviceModelStorageInterface

File: modules/OCPP201/doc/sequence_config_service_and_ocpp.puml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+!pragma teoz true
+participant CSMS order 10
+participant libocpp order 20
+participant ComposedDeviceModel order 30
+database InternalStorage order 40
+database ExternalStorage order 50
```
Comment (Pietfried) [inline modules/OCPP201/doc/sequence_config_service_and_ocpp.puml:8]: Looks like 
```suggestion
database DeviceModelStorageSqlite order 40
database EverestDeviceModelStorage order 50
```
would be more appropriate

**Review by Pietfried — CHANGES_REQUESTED** [review]
lgtm! Requested some documentation changes partially caused by the renaming to `DeviceModelStorageInterface`

**Review by Pietfried — APPROVED** [review]

**Comment by Pietfried** [comment]
:+1: 

please merge https://github.com/EVerest/libocpp/pull/768 and update the `dependencies.yaml` in this PR afterwards

**Pull request merged**